src/app/modules/timesheet/services/timesheet.service.spec.ts

```diff
@@ -61,6 +60,34 @@ describe('TimesheetService', () => {
     });
   });
 
+  fdescribe('getTransferToken())', () => {
+    let transferToken;
+    let token;
+    beforeEach(() => {
+      service.timesheet.invoice = Object.assign({}, new Invoice(), {
+          number: '458789',
+          provider: new Company('Rémy dupont'),
+          client: new Company('Iteatime'),
+        });
+      transferToken = btoa(unescape(encodeURIComponent(JSON.stringify({
+          mode: 'edit',
+          timesheet: Object.assign({ ...service.timesheet }, {
```

Comment:
I put the `object.assign` commentary only one time, but careful for all other, it's always better to avoid mutations of objects